pallets/api/src/fungibles/tests.rs

```diff
@@ -675,3 +675,140 @@ mod read_weights {
 		assert_eq!(token_exists.proof_size(), 3675);
 	}
 }
+
+mod ensure_codec_indexes {
+	use super::{Encode, RuntimeCall, *};
+	use crate::{fungibles, fungibles::Call::*, mock::RuntimeCall::Fungibles};
+
+	#[test]
+	fn ensure_read_variant_indexes() {
+		[
+			// explicit u8 to help Rust infer the type
+			(TotalSupply::<Test>(Default::default()).encode(), 0u8, "TotalSupply"),
+			(
+				BalanceOf::<Test> { token: Default::default(), owner: Default::default() }.encode(),
+				1,
+				"BalanceOf",
+			),
+			(
+				Allowance::<Test> {
+					token: Default::default(),
+					owner: Default::default(),
+					spender: Default::default(),
+				}
+				.encode(),
+				2,
+				"Allowance",
+			),
+			(TokenName::<Test>(Default::default()).encode(), 8, "TokenName"),
+			(TokenSymbol::<Test>(Default::default()).encode(), 9, "TokenSymbol"),
+			(TokenDecimals::<Test>(Default::default()).encode(), 10, "TokenDecimals"),
+			(TokenExists::<Test>(Default::default()).encode(), 18, "TokenExists"),
+		]
+		.iter()
+		.for_each(|(encoded_variant, expected_index, name)| {
+			assert_eq!(encoded_variant[0], *expected_index, "{name} variant index changed");
+		})
+	}
+
+	#[test]
+	fn ensure_dispatchable_indexes() {
+		use fungibles::Call::*;
+
+		[
+			(
+				Fungibles(transfer {
+					token: Default::default(),
+					to: Default::default(),
+					value: Default::default(),
+				}),
+				3u8, // explicit u8 to help Rust infer the type
```

Comment:
```suggestion
				3u8,
```
As above, your call.